communitiesuk/funding-service-design-assessment-store
# Fs 3022 generalise populate location data for other funds

https://digital.dclg.gov.uk/jira/browse/FS-3022
https://digital.dclg.gov.uk/jira/browse/FS-3023

### Description
- Refactored `get_postcode_from_questions` function to work with other funds
- Implemented location extraction in the `derive_application_values` function part of import script(so for future imports no need to run `populate_location_data` script)
- Make location data consistent to `Not Available` when not found
- Improved the import & populate script working with shortnames (see below How to test section)
- Made import script resilient by logging the failed application_id and continuing the import ([FS-3023](https://digital.dclg.gov.uk/jira/browse/FS-3023))
- Updated read me file



### How to test
_If manual testing is needed, give suggested testing steps_

#### To test import script
`docker exec -it <container_id> python -m scripts.import_from_application --roundid=c603d114-5364-4474-a0c4-c41cbf4d3bbd --app_type=COF`
or with short name
`docker exec -it <container_id> python -m scripts.import_from_application --fundround=COFR3W1`

#### To test populate location data script
`docker exec -ti <container_id> scripts/populate_location_data.py --fund_id 47aef2f5-3fcb-4d45-acb5-f0152b5f03c4 --round_id c603d114-5364-4474-a0c4-c41cbf4d3bbd --update_db True --write_csv False`
or with short name
`docker exec -ti <container_id> scripts/populate_location_data.py --fundround COFR2W3 --update_db True --write_csv False`

Alternatively, if using VsCode select the relevant debugger to test( see [here](https://github.com/communitiesuk/funding-service-design-assessment-store#launch-configurations-in-vscode)) 

### Screenshots of UI changes (if applicable)
